apps/server/src/providers/index.test.ts

```diff
@@ -0,0 +1,114 @@
+import { http, HttpResponse } from 'msw'
+import { server } from '../test/testServer'
+import { ProviderApiBase } from './index'
+import { createClient } from '../__mocks__/redis'
+import { ConnectionStatus, OAuthStatus } from '../../src/shared/contract'
+import { READ_MEMBER_STATUS_PATH } from '../test/handlers'
+import { MxApi } from './mx'
+
+const redisMock = createClient()
+const testConnectionId = 'test_connection_id'
+
+const mxApi = new MxApi(
+  {
+    mxProd: {
+      username: 'testUsername',
+      password: 'testPassword'
+    },
+    storageClient: redisMock
+  },
+  false
+)
+
+const providerApiBase = new ProviderApiBase({
+  context: {
+    provider: 'mx',
+    auth: {
+      token: 'test_token',
+      iv: 'test_iv'
+    },
+    resolved_user_id: 'test_user_id'
+  }
+})
+
+describe('ProviderApiBase', () => {
+  describe('getOauthState', () => {
+    beforeAll(async () => {
+      await providerApiBase.init()
+      providerApiBase.serviceClient = mxApi
+    })
+
+    describe('oauth status PENDING', () => {
```

Comment:
Typically I use describe for functions, and it for tests cases for those functions. so it("returns a pending oauth state"